examples/ssr/src/instantsearch.js

```diff
@@ -0,0 +1,106 @@
+import instantsearch from 'instantsearch.js/es/index';
+import algoliaHelper from 'algoliasearch-helper';
+
+export const createInstantSearch = ({ searchClient, indexName, options }) => {
+  const search = instantsearch({
+    searchClient,
+    indexName,
+    ...options,
+  });
+
+  search._isSsr = true;
+
+  search.ssr = async params => {
+    search.helper = algoliaHelper(searchClient, indexName, {
+      // parameters set by default
+      highlightPreTag: '__ais-highlight__',
+      highlightPostTag: '__/ais-highlight__',
+      ...params,
+    });
+
+    const { content: lastResults } = await search.helper.searchOnce();
+
+    // TODO: maybe use the life cycle version?
+    search.helper.lastResults = lastResults;
+
+    return {
+      ais: {
+        lastResults: JSON.parse(JSON.stringify(lastResults)),
+      },
+    };
+  };
+
+  search.giveRenderPropsServerOrSomething = widget => {
+    widget.init({
+      state: search.helper.lastResults._state,
+      helper: search.helper,
+      templatesConfig: {},
+      createURL: () => '#',
+      onHistoryChange: () => {},
+      instantSearchInstance: search,
+    });
+
+    widget.render({
+      state: search.helper.lastResults._state,
+      results: search.helper.lastResults,
+      helper: search.helper,
+      templatesConfig: {},
+      createURL: () => '#',
+      instantSearchInstance: search,
+      searchMetadata: {
+        isSearchStalled: false,
+      },
+    });
+  };
+
+  search.giveRenderPropsClientOrSomething = widget => {
+    widget.render({
+      state: search.helper.lastResults._state,
+      results: search.helper.lastResults,
+      helper: search.helper,
+      templatesConfig: {},
+      createURL: () => '#',
+      instantSearchInstance: search,
+      searchMetadata: {
+        isSearchStalled: false,
+      },
+    });
+  };
+
+  search.injectOrHydrate = () => {
```

Comment:
frontend: read from serialised state to set the correct state in the InstantSearch instance